### PR TITLE
feat: implement cross-chain mechanism between op and greenfield

### PR DESCRIPTION
**Pull request by yutianwu**
### Description

This pr implements the cross-chain mechanism between op and greenfield.

### Rationale

Implement the cross-chain mechanism between op and greenfield.

### Example

N/A

### Changes

Notable changes:
* implement the cross-chain mechanism between op and greenfield

### CODE REVIEW

File: x/crosschain/keeper/keeper.go [hunk 1]
```diff
@@ -175,7 +175,13 @@ func (k Keeper) RegisterChannel(name string, id sdk.ChannelID, app sdk.CrossChai
 
 // IsDestChainSupported returns the support status of a dest chain
 func (k Keeper) IsDestChainSupported(chainID sdk.ChainID) bool {
-	return chainID == k.cfg.destBscChainId
+	if chainID == k.cfg.destBscChainId {
```
Comment (guagualvcha) [inline x/crosschain/keeper/keeper.go:178]: simplify : `return  chainID == k.cfg.destBscChainId || k.cfg.destOpChainId == chainID`

Comment (guagualvcha) [reply x/crosschain/keeper/keeper.go:178]: As `opBNB` will not be supported on day1, I am afraid that the user may send such a tx on gnfd. Can we use a hardfork to disable it or some other way?

File: x/gov/types/crosschain.go [hunk 2]
```diff
@@ -6,9 +6,13 @@ import (
...
 
 const (
-	SyncParamsChannel                 = "syncParametersChange"
-	SyncParamsChannelID sdk.ChannelID = 3
-	KeyUpgrade                        = "upgrade"
+	SyncParamsChannel   = "syncParametersChange"
+	SyncSequenceChannel = "syncSequence"
+
+	SyncParamsChannelID   sdk.ChannelID = 3
+	SyncSequenceChannelID sdk.ChannelID = 7
```
Comment (guagualvcha) [inline x/gov/types/crosschain.go:13]: `SyncSequenceChannelID`. is no longer needed.

File: x/gov/types/crosschain.go [hunk 3]
```diff
@@ -6,9 +6,13 @@ import (
 )
 
 const (
-	SyncParamsChannel                 = "syncParametersChange"
-	SyncParamsChannelID sdk.ChannelID = 3
-	KeyUpgrade                        = "upgrade"
+	SyncParamsChannel   = "syncParametersChange"
+	SyncSequenceChannel = "syncSequence"
```
Comment (guagualvcha) [inline x/gov/types/crosschain.go:10]: No longer needed.

**Review by alexgao001 — COMMENTED** [review]
Seems ChainID is missing in query sequence/crosschain package  request proto?

**Review by guagualvcha — APPROVED** [review]

**Review by alexgao001 — APPROVED** [review]

**Pull request merged**